ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -353,7 +354,8 @@ void onNext(AppendEntriesRequestProto proto)
       }
       // stall for stream to be ready.
       while (!stream.isReady() && running) {
-        sleep(waitForReady, isHeartBeat);
+        //sleep(waitForReady, isHeartBeat);
+        LockSupport.parkNanos(waitForReady.toLong(TimeUnit.NANOSECONDS));
```

Comment:
@jojochuang , thanks for the update!  Sorry that I missed the new change earlier.

How about changing it in the sleep(..) method?
```java
@@ -408,12 +409,9 @@ public class GrpcLogAppender extends LogAppenderBase {
 
   private static void sleep(TimeDuration waitTime, boolean heartbeat)
       throws InterruptedIOException {
-    try {
-      waitTime.sleep();
-    } catch (InterruptedException e) {
-      Thread.currentThread().interrupt();
-      throw IOUtils.toInterruptedIOException(
-          "Interrupted appendLog, heartbeat? " + heartbeat, e);
+    LockSupport.parkNanos(waitTime.toLong(TimeUnit.NANOSECONDS));
+    if (Thread.currentThread().isInterrupted()) {
+      throw new InterruptedIOException("Interrupted appendLog, heartbeat? " + heartbeat);
     }
   }
```